packages/core/src/use-query.ts

```diff
@@ -6,9 +6,19 @@ type RestQueryResult<T extends QueryFn> = Omit<QueryResult<PromiseReturnType<T>>
 
 export function useQuery<T extends QueryFn>(
   queryFn: T,
-  params?: InferUnaryParam<T>,
+  params: InferUnaryParam<T>,
   options?: QueryOptions<T>,
```

Comment:
unrelated and I don't want to hold up this PR, but I dug into this for a sec and it seems like QueryOptions<T> expects T to be the type of the result of the queryFn, not the queryFn itself.

Reply:
You are right! I just pushed a fix for this